lib/core/Error.ts

```diff
@@ -4,10 +4,20 @@
  */
 export class SidetreeError extends Error {
   constructor (public code: string, message?: string) {
-    super(message ? message : code);
+    super(message ? `${code}: ${message}` : code);
 
     // NOTE: Extending 'Error' breaks prototype chain since TypeScript 2.1.
     // The following line restores prototype chain.
     Object.setPrototypeOf(this, new.target.prototype);
   }
+
+  /**
+   * Returns a new SidetreeError object using the inputs.
+   *
+   * @param code The error code.
+   * @param err The error exception thrown.
+   */
+  public static createFromError (code: string, err: Error): SidetreeError {
+    return new SidetreeError(code, err ? err.message : undefined);
```

Comment:
Sanity: it seems not possible for `err` to be `undefined`?

Reply:
Could be undefined at runtime. For example if the Promise is rejected without an error (which is a bug that I found). So keeping this.